### DISCUSSION TITLE
Lazy

### DISCUSSION THREAD

**Discussion by krahabb** [krahabb/meross_lan#248] [Releases]
The 'Lazy' release is about to be finished and I would like to have some feedback from you 'early adopters' which managed to install the pre-release versions (alpha and beta) available in the repo. 
This discussion aims to collect both positive feedback (confirming the release works) and the negative one. If you have a 'kind of an issue' feel free to also open a standard issue for the matter so it could be tracked in its own thread but if it's just minor or something which could be improved let's track it here
Nevertheless, if you can confirm any or all of your devices are happily working with these pre-release versions please submit which devices/firmwares are correctly working in order to help me feel confident and speed up the release process for the final stable version

Thank you!

**Comment by nao-pon** [comment]
I have confirmed that it works fine on multiple devices below.

- Smart Plug MSS110 2.0.0 FW: 2.1.20
- Smatt Plug MSS110 5.0.0 FW: 5.1.1

All devices are connected to a local MQTT broker by tools at https://github.com/bytespider/Meross.

**Comment by Duke-Box** [comment]
I can confirm it works on MSG100 Garage Door Opener F/W: 3.2.7

**Comment by PixelPusher247** [comment]
It is working fine for me connected to a local MQTT broker with the following devices:
- 8x Smart Plug MSS310 FW: 6.1.10
- 7x Smart Plug MSS210 FW: 2.2.2
- 3x Smart Plug MSS210 FW: 2.1.7
- 4x Smart Plug MSS110 FW: 6.2.5

**Comment by PixelPusher247** [comment]
One question/suggestion: 
Would it be possible to log the device name as configured in the Home Assistant integrations page when logging which devices timestamp are falling behind? 
Right now the message looks something like: `MerossDevice({UUID}) has incorrect timestamp: 14 seconds behind HA`. I always have to check the MAC address of the device to find which one it is, would be nice if it would show the friendly name 😃

**Reply by krahabb** [reply]
Sure..fixed before official release.
Thank you for the nice suggestion..the uuid are a bit cumbersome yeah

**Comment by Fuchschr11** [comment]
> The 'Lazy' release is about to be finished and I would like to have some feedback from you 'early adopters' which managed to install the pre-release versions (alpha and beta) available in the repo. This discussion aims to collect both positive feedback (confirming the release works) and the negative one. If you have a 'kind of an issue' feel free to also open a standard issue for the matter so it could be tracked in its own thread but if it's just minor or something which could be improved let's track it here Nevertheless, if you can confirm any or all of your devices are happily working with these pre-release versions please submit which devices/firmwares are correctly working in order to help me feel confident and speed up the release process for the final stable version
> 
> Thank you!

Hi, update with latest HA and HACS is not working. Progress spins forever... what to check / do?

Log says:

Logger: homeassistant.config_entries
Source: config_entries.py:816 
First occurred: 20:15:40 (11 occurrences) 
Last logged: 20:15:40

Error occurred loading configuration flow for integration meross_lan: cannot import name 'MerossApi' from 'custom_components.meross_lan' (unknown location)

Thank you.

**Reply by krahabb** [reply]
Hello @Fuchschr11,
It might be the download through HACS failed or got corrupted somehow. Could you try re-installing?

**Comment by Fuchschr11** [comment]
Thank you, but all other updates work in HACS.

Will I loose my meross devices when deleting HACS and reinstalling.
What is the exact procedure for this?

Greetings
Christian

Von: krahabb ***@***.***>
Datum: Sonntag, 5. Februar 2023 um 22:51
An: krahabb/meross_lan ***@***.***>
Cc: Fuchschr11 ***@***.***>, Mention ***@***.***>
Betreff: Re: [krahabb/meross_lan] Lazy (Discussion #248)

Hello @Fuchschr11<https://github.com/Fuchschr11>,
It might be the download through HACS failed or got corrupted somehow. Could you try re-installing?

**Reply by krahabb** [reply]
Hello @Fuchschr11,
No need to re-install HACS (that would be a pain!) just using HACS re-download meross_lan without uninstalling it

**Comment by Fuchschr11** [comment]
Thank you,
you mean here to Click „ReDownload“.[Ein Bild, das Text enthält.  Automatisch generierte Beschreibung]

Will not work either, spins very long and never ends.


Von: krahabb ***@***.***>
Datum: Dienstag, 7. Februar 2023 um 13:09
An: krahabb/meross_lan ***@***.***>
Cc: Fuchschr11 ***@***.***>, Mention ***@***.***>
Betreff: Re: [krahabb/meross_lan] Lazy (Discussion #248)

Hello @Fuchschr11<https://github.com/Fuchschr11>,
No need to re-install HACS (that would be a pain!) just using HACS re-download meross_lan without uninstalling

**Reply by krahabb** [reply]
Yep, 'Redownload' will do..but if it spins forever there are some connectivity issues...
I've just tried on my production instance and it works: spins 5-10 seconds and then 'Done'

I'm not that expert about HACS though: it usually caches an access token in order to query github repos so it might be the token has expired or other kind of issues..not sure if it provides any deeper log but surely the issue lies in HACS not being able to correctly download the repo
